### ISSUE TITLE
Allow learning rates of specific variables to be scaled

### ISSUE THREAD

**Issue by alexatknit** [tensorflow/tensorflow#313]
When repurposing a network, it can be useful to set layer-wise learning rates so that the final layer (which has random weights) does most of the learning initially. A useful pattern for applying learning rate scaling would be tf.stop_gradient(input, name=None), which can be seen as effectively scaling the learning rate by a factor of zero (obviously it also stops computation).

**Comment by vrv** [comment]
You can use different optimizers for any subset of variables in your graph to get per-layer learning rates.

The 'minimize' function of an optimizer http://www.tensorflow.org/api_docs/python/train.html#optimizers allows you to specify which variables should be updated.  Let us know if this isn't what you need ...

**Comment by alexatknit** [comment]
nope, its perfect, thanks

**Comment by ascust** [comment]
hi guys, I would like to know more about this layer-wise learning rate since I am using this technique for fine-tuning. It seems by using stop_gradient, we can only be able to specify which layer should be prevented from updating. What if I want to give low level layers some low learning rate like 0.0001 and give high level layers high value like 0.01. How am I supposed to do this. Thank you

**Comment by vrv** [comment]
This should really be posted on StackOverflow -- you'll get more community help there :)
